### PR TITLE
add auth interface and implemention

### PR DESCRIPTION
**Pull request by StupidHod**
add simple implementation for auth by convert gRPC interface

### CODE REVIEW

File: src/main/java/com/coreos/jetcd/EtcdAuthImpl.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ * Implementation of etcd auth client
+ */
+public class EtcdAuthImpl implements EtcdAuth{
+    
+    private ManagedChannel channel;
+    private AuthGrpc.AuthFutureStub authStub;
+    
+    
+    public EtcdAuthImpl(ManagedChannel channel){
+        this.channel=channel;
```
Comment (AdoHe) [inline src/main/java/com/coreos/jetcd/EtcdAuthImpl.java]: could we directly pass stub here?

**Comment by AdoHe** [comment]
Generally LGTM. We may need to consider how to support connect to TLS secured etcd cluster or even support basic auth.

File: src/main/java/com/coreos/jetcd/EtcdAuthImpl.java (hunk 1 above)
Comment (StupidHod) [inline src/main/java/com/coreos/jetcd/EtcdAuthImpl.java]: @AdoHe Tks, I will construct it directly by stub.

**Comment by AdoHe** [comment]
Also please help add auth tests once we resolve testng issue @StupidHod .

**Comment by StupidHod** [comment]
@AdoHe Ok, I have add some test in my local repositorie, wait for the resolution.

File: src/main/java/com/coreos/jetcd/EtcdAuth.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    ListenableFuture<AuthEnableResponse> authEnable();
+
+    ListenableFuture<AuthDisableResponse> authDisable();
+
+    // ***************
+    // User Manage
+    // ***************
+
+    ListenableFuture<AuthUserAddResponse> userAdd(ByteString name, ByteString passWord);
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/EtcdAuth.java]: password instead of passWord

File: src/main/java/com/coreos/jetcd/EtcdAuth.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    ListenableFuture<AuthRoleDeleteResponse> roleDelete(ByteString role);
+
+
+    /**
+     * Authentication with user name and password, the service will return token if authentication succeeds
+     * @param name
+     * @param passWord
+     * @return AuthenticateResponse with token or error information
+     */
+    ListenableFuture<AuthenticateResponse> authenticate(ByteString name, ByteString passWord);
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/EtcdAuth.java]: let's remove authenticate for now. Ideally, we should call authenticate internally inside the client library and setup authenticated connection for users. Users do not need to do this themselves. Or they have to attach the token to each requests, which is tedious.

**Comment by StupidHod** [comment]
@xiang90 Tks, I have correct the word error and remove the authenticate function as your advice.

**Comment by xiang90** [comment]
lgtm. Squash commits into one please.

**Comment by xiang90** [comment]
@StupidHod Thanks a lot!

**Pull request merged**